Lombiq.Hosting.Tenants.EnvironmentRobots/Extensions/IHostEnvironmentEnvironmentRobotsExtensions.cs

```diff
@@ -0,0 +1,12 @@
+using Lombiq.Hosting.Tenants.EnvironmentRobots.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Lombiq.Hosting.Tenants.EnvironmentRobots.Extensions;
+public static class IHostEnvironmentEnvironmentRobotsExtensions
+{
+    public static bool IsProductionWithConfiguration(
+        this IHostEnvironment hostEnvironment,
+        IOptions<EnvironmentRobotsOptions> options) =>
+        options.Value.IsProduction ?? hostEnvironment.IsProduction();
```

Comment:
`EnvironmentRobotsOptions.IsProduction` does not have to be `nullable` if you replace `??` with `||` here.

Reply:
It has to be `nullable`. Zoltán already mentioned this: 
> in case ASP.NET Core should be told it runs in Production but not actually (like in staging, that should be as close to production as possible).

For example in staging, if it's not nullable and we are using `||` here it would return `true` every time here, because of ,  `hostEnvironment.IsProduction();` even if we use the custom configuration and set it to false.

Reply:
Wow! Good catch. Thanks for the explanation.

Reply:
Please approve if you feel that the conversation is resolved.